### PR TITLE
Don't send null values

### PR DESCRIPTION
**Pull request by d4rken**
Overall goal: Don't write null values into the output stream. See #86 
To help with that most methods that produce output for the stream have been made static and self contained. The helper methods and parameters (where applicable) have been annotated to indicate whether the value can be null using Androids annotation library.
Fields holding generated data have also been made final (was suggested by LINT checks).

Android Studio also seems to have corrected some javadoc formatting which was not intended, but oh well...

This also includes the same fixes as PR #92

### CODE REVIEW

**Comment by d4rken** [comment]
Doesn't look like a code issue:

```
* What went wrong:

Execution failed for task ':connectedDebugAndroidTest'.

> com.android.builder.testing.api.TestException: com.android.builder.testing.api.DeviceException: com.android.ddmlib.ShellCommandUnresponsiveException
```

File: src/main/java/com/bugsnag/android/Client.java [hunk 1]
```diff
@@ -287,22 +287,22 @@ public void setUserName(String name) {
...
-     *
+     * <p/>
      * For example:
+     * <p/>
+     * client.beforeNotify(new BeforeNotify() {
+     * public boolean run(Error error) {
+     * error.setSeverity(Severity.INFO);
+     * return true;
+     * }
+     * })
```
Comment (kattrali) [inline src/main/java/com/bugsnag/android/Client.java]: This formatting probably needs corrected to the original version. Maybe Android Studio would like a `<code>` block better.

Comment (JakeWharton) [inline src/main/java/com/bugsnag/android/Client.java]: Multi-line code snippets in Javadoc should use:
```
<pre>{@code
blah blah
}</pre>
```

**Comment by kattrali** [comment]
The changes look good. I got a bit bogged down with the unrelated changes, but looked over everything. :) Next up is doing more integrated testing, just for sanity checks. Thanks for putting work into this, @d4rken.

File: src/main/java/com/bugsnag/android/Client.java (hunk 1 above)
Comment (kattrali) [inline src/main/java/com/bugsnag/android/Client.java]: Ah, that's what I was looking for. Thanks, @JakeWharton. :+1:

**Pull request merged**